pkg/proc/target.go

```diff
@@ -144,15 +145,26 @@ type NewTargetConfig struct {
 
 // DisableAsyncPreemptEnv returns a process environment (like os.Environ)
 // where asyncpreemptoff is set to 1.
-func DisableAsyncPreemptEnv() []string {
-	env := os.Environ()
+func DisableAsyncPreemptEnv(env []string) {
 	for i := range env {
 		if strings.HasPrefix(env[i], "GODEBUG=") {
 			// Go 1.14 asynchronous preemption mechanism is incompatible with
 			// debuggers, see: https://github.com/golang/go/issues/36494
 			env[i] += ",asyncpreemptoff=1"
 		}
 	}
+}
+
+// MergeInheritedEnviron merges given environment variables with inherited ones,
+// in the form "key=value".
+func MergeInheritedEnviron(environ []string) []string {
```

Comment:
I think it's weird that this doesn't work like the `Env` field of `os/exec.Cmd`, i.e. when it's nil it uses `os.Environ()` otherwise it uses whatever it's passed.